crates/sc-proof-of-time/src/lib.rs

```diff
@@ -85,3 +88,94 @@ impl BootstrapParams {
         }
     }
 }
+
+/// Interface to consensus.
+pub trait PotConsensusState<Block: BlockT>: Send + Sync {
+    /// Called by consensus when trying to claim the slot.
+    /// Returns the proofs in the slot range
+    /// [parent.last_proof.slot + 1, slot_number - global_randomness_reveal_lag_slots].
+    fn get_block_proofs(
+        &self,
+        slot_number: SlotNumber,
+        block_number: NumberFor<Block>,
+        parent_block_proofs: &[PotProof],
+    ) -> Result<Vec<PotProof>, PotConsensusError>;
+
+    /// Called during block import validation.
+    /// Verifies the sequence of proofs in the block being validated.
+    fn verify_block_proofs(
+        &self,
+        slot_number: SlotNumber,
+        block_number: NumberFor<Block>,
+        block_proofs: &[PotProof],
+        parent_block_proofs: &[PotProof],
+    ) -> Result<(), PotConsensusError>;
+}
+
+#[derive(Debug, thiserror::Error)]
```

Comment:
We generally use `#![warn(missing_docs)]` to remind us about us about comments on public data structures, including errors. And like with others, move to where they are used. It looks odd to have config and error in one file and implementation that uses them in another.

Reply:
I don't think we need all the stuff in `lib.rs`, in fact it is preferable to make modules public rather than re-export everything in a flat list. It helps with discovery and generated docs make much more sense when related things are grouped into logical modules and not all at the root.

Reply:
Now lib.rs has things which are exported and don't belong anywhere else